libraries/Microsoft.Bot.Builder.Dialogs/Templating/MessageActivityTemplateGenerator.cs

```diff
@@ -0,0 +1,360 @@
+        }
+
+        private IMessageActivity CreateActivityFromText(IMessageActivity activity, string text)
+        {
+            activity.Text = text.Trim();
+            activity.Speak = text.Trim();
+            return activity;
+        }
+
+        private IMessageActivity CreateActivityFromJObj(IMessageActivity activity, JObject lgJObj)
+        {
+            var type = GetTemplateType(lgJObj);
+
+            switch (type)
+            {
+                case "Herocard":
+                    AddGenericCardAtttachment(activity, HeroCard.ContentType, lgJObj);
+                    break;
+
+                case "ThumbnailCard":
+                    AddGenericCardAtttachment(activity, ThumbnailCard.ContentType, lgJObj);
+                    break;
+
+                case "AudioCard":
+                    AddGenericCardAtttachment(activity, AudioCard.ContentType, lgJObj);
+                    break;
+
+                case "VideoCard":
+                    AddGenericCardAtttachment(activity, VideoCard.ContentType, lgJObj);
+                    break;
+
+                case "AnimationCard":
+                    AddGenericCardAtttachment(activity, AnimationCard.ContentType, lgJObj);
+                    break;
+
+                case "SigninCard":
+                    AddGenericCardAtttachment(activity, SigninCard.ContentType, lgJObj);
+                    break;
+
+                case "OAuthCard":
+                    AddGenericCardAtttachment(activity, OAuthCard.ContentType, lgJObj);
+                    break;
+
+                case "AdaptiveCard":
+                    // json object
+                    AddJsonAttachment(activity, "application/vnd.microsoft.card.adaptive", lgJObj);
+                    break;
+
+                case "Activity":
+                    BuildNormalActivity(activity, lgJObj);
+                    break;
+
+                default:
+                    activity.Text = lgJObj.ToString();
+                    break;
+            }
+
+            return activity;
+        }
+
+        private void BuildNormalActivity(IMessageActivity activity, JObject lgJObj)
+        {
+            foreach (var item in lgJObj)
+            {
+                var property = item.Key.Trim().ToLower();
+                var value = item.Value;
+
+                switch (property.ToLower())
+                {
+                    case "text":
+                        activity.Text = value.ToString();
+                        break;
+
+                    case "speak":
+                        activity.Speak = value.ToString();
+                        break;
+
+                    case "inputhint":
+                        activity.InputHint = value.ToString();
+                        break;
+
+                    case "attachments":
+                        activity.Attachments = GetAttachments(value);
+                        break;
+
+                    case "suggestedActions":
```

Comment:
This can never be hit due to the ".ToLower()".

Reply:
sure. Fix